app/components/AddressLink.tsx

```diff
@@ -0,0 +1,58 @@
+'use client';
+
+import { mainnetConfigForENS } from '@/providers/chain-config';
+import { shortenAddress } from '@/utils';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useMemo } from 'react';
+import { twMerge } from 'tailwind-merge';
+import { Address, getAddress, http } from 'viem';
+import { mainnet } from 'viem/chains';
+import { useEnsName } from 'wagmi';
+
+interface AddressLinkProps {
```

Comment:
Let `AddressLinkProps` extend from `Link` props, and spread the rest of the props to the link component. This to allow passing link props to the component.